clang/lib/Driver/ToolChains/AMDGPU.cpp

```diff
@@ -648,6 +648,15 @@ void amdgpu::Linker::ConstructJob(Compilation &C, const JobAction &JA,
         Args.MakeArgString("-plugin-opt=-mattr=" + llvm::join(Features, ",")));
   }
 
+  if (Args.hasArg(options::OPT_gpustartfiles)) {
```

Comment:
Is there prior art for a flag to link crt? (i.e. can we just use that instead of inventing a new -gpu flag)

Reply:
Unfortunately not, there's `-nostartfiles` and `-nostdlib`, but nothing to invert that. Adding one would probably add a few hundred changes of `hasArg` to `hasFlag`. The other targets just assume they're hosted, and I can't just link calls to `main` by default since no one wants that unless they're trying to write tests or something.

Reply:
can we make that have a positive pair, like other flags? No gpu prefix?

Reply:
What I meant, but that's a much bigger change. Maybe I could get away with only using the positive variant in the GPU toolchain?

Reply:
Default value would be a toolchain choice, so yes?

Reply:
Done.